Source/Diagnostics/WarpXOpenPMD.cpp

```diff
@@ -795,7 +795,8 @@ WarpXParticleCounter::GetParticleOffsetOfProcessor(const long& numParticles,
     amrex::ParallelGather::Gather (numParticles, result.data(), -1, amrex::ParallelDescriptor::Communicator());
 
     sum = 0;
-    for (int i=0; i<result.size(); i++) {
+    int const num_results = result.size();
+    for (int i=0; i<num_results; i++) {
```

Comment:
Why do we need a const variable to store the size and then use in the for loop instead of directly using result.size() in the for loop?

Reply:
I touched this line primarily because it had a signed-unsigned comparison. While I was on it, I just decided to evaluate `.size()` once instead of running it on each loop.

This is super marginal here, but generally will aide the compiler to vectorize your code. THis is because `std::vector::size()` is actually doing a computation:
```
// snippet from glibc/libstdc++ on my Ubuntu 18.04

/**  Returns the number of elements in the %vector.  */
      size_type
      size() const _GLIBCXX_NOEXCEPT
      { return size_type(this->_M_impl._M_finish - this->_M_impl._M_start); }
```

It's maginal since it's an outer loop, the compiler probably recognizes `result` is constant in it and optimizes, but since I had to cast anyway from `size_t` to `int` I thought I just do it explicitly.

Reply:
Ah thank you!
@ax3l